pkg/ebpf/c/vmlinux.h

```diff
@@ -791,7 +780,76 @@ struct ethhdr {
 
 typedef __u16 __sum16;
 
-// TODO: can't CO-RE iphdr (check)
+enum kernel_read_file_id
+{
+    READING_UNKNOWN = 0,
+    READING_FIRMWARE = 1,
+    READING_MODULE = 2,
+    READING_KEXEC_IMAGE = 3,
+    READING_KEXEC_INITRAMFS = 4,
+    READING_POLICY = 5,
+    READING_X509_CERTIFICATE = 6,
+    READING_MAX_ID = 7,
+};
+
+struct kretprobe_instance {
+};
+typedef int kprobe_opcode_t;
+struct kprobe;
+
+typedef int (*kprobe_pre_handler_t)(struct kprobe *, struct pt_regs *);
+typedef void (*kprobe_post_handler_t)(struct kprobe *, struct pt_regs *, unsigned long flags);
+typedef int (*kretprobe_handler_t)(struct kretprobe_instance *, struct pt_regs *);
+
+struct kprobe {
+    kprobe_opcode_t *addr;
+    const char *symbol_name;
+    kprobe_pre_handler_t pre_handler;
+    kprobe_post_handler_t post_handler;
+};
+
+struct seq_file {
+};
+
+struct seq_operations {
+    void *(*start)(struct seq_file *m, loff_t *pos);
+    void (*stop)(struct seq_file *m, void *v);
+    void *(*next)(struct seq_file *m, void *v, loff_t *pos);
+    int (*show)(struct seq_file *m, void *v);
+};
+
+enum bpf_attach_type
+{
+    BPF_CGROUP_INET_INGRESS = 0,
+    BPF_CGROUP_INET_EGRESS = 1,
+};
+
+enum event_type
+{
+    EVENT_CGROUP_SKB_INGRESS = 1,
+    EVENT_CGROUP_SKB_EGRESS,
+    EVENT_KPROBE_SOCK_ALLOC_FILE,
+    EVENT_KRETPROBE_SOCK_ALLOC_FILE,
+    EVENT_KPROBE_CGROUP_BPF_FILTER_SKB,
+    EVENT_KRETPROBE_CGROUP_BPF_FILTER_SKB,
+};
+
+enum bpf_hdr_start_off
+{
+    BPF_HDR_START_MAC = 0,
+    BPF_HDR_START_NET = 1,
+};
```

Comment:
and these?

Reply:
![image](https://user-images.githubusercontent.com/7395852/194959144-5fce8268-cfef-4a61-b3ae-e5cb6d0e9428.png)